kore/app/repl/Main.hs

```diff
@@ -0,0 +1,177 @@
+{-# OPTIONS_GHC -Wno-unused-top-binds    #-}
+-- Added for outputFileName
+
+module Main (main) where
+
+import Control.Applicative
+       ( optional )
+import Data.Semigroup
+       ( (<>) )
+import Options.Applicative
+       ( InfoMod, Parser, argument, auto, fullDesc, header, help, long,
+       metavar, option, progDesc, readerError, str, strOption, value )
+
+import Data.Limit
+       ( Limit (..) )
+import Kore.AST.Sentence
+       ( ModuleName (..) )
+import Kore.Exec
+       ( proveWithRepl )
+import Kore.Logger.Output
+       ( emptyLogger )
+import Kore.Step.Simplification.Data
+       ( evalSimplifier )
+
+import           GlobalMain
+import qualified SMT as SMT
+
+-- | Represents a file name along with its module name passed.
+data KoreModule = KoreModule
+    { fileName   :: !FilePath
+    , moduleName :: !ModuleName
+    }
+
+-- | SMT Timeout and (optionally) a custom prelude path.
+data SmtOptions = SmtOptions
+    { timeOut :: !SMT.TimeOut
+    , prelude :: !(Maybe FilePath)
+    }
+
+-- | Options for the kore repl.
+data KoreReplOptions = KoreReplOptions
+    { definitionModule :: !KoreModule
+    , specModule       :: !KoreModule
+    , smtOptions       :: !SmtOptions
+    , outputFileName   :: !(Maybe FilePath)
```

Comment:
Since we don't plan to do anything with this option for the REPL, I would rather omit the field. We can still have the parser accept and ignore the option by joining the option parser with `<*` instead of `<*>`.